aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -2344,37 +2344,81 @@ void fake_quantize_learnable_tensor_grad_kernel_cpu(
   });
 }
 
+template <typename SelfType>
+void _fake_quant_per_channel_cachemask_cpu_helper(
+    TensorIterator& iter,
+    TensorIterator& iter_mask,
+    const int64_t quant_min,
+    const int64_t quant_max) {
+
+  const auto& zero_point_dtype = iter.input_dtype(2);
+
+  if(at::isFloatingType(zero_point_dtype)){
+    // When zero_point is float, quantize mirroring affine quantizer equation
+    // Xq = Round(Xf * inv_scale + zero_point)
```

Comment:
So in the end, the `Xq` is also floating number? should it be cast to int?

Reply:
Yes, this is done at L2364 and L2374 via `std::lrintf`

Reply:
Can you clarify with a code-pointer where you would expect the int cast'ing to happen?  I think this should be handled with `std::lrintf`, then when "fake" de-quantizing, should be a float type again.

Reply:
oh, I was referring to the fake quantized `x` (`xq` here), normally aren't we using the int type as the quantized value? 

But I think it should be good here, as the kernel fake quant then dequant here already